js/interactive-guides/microprofile-config/microprofile-config-callback.js

```diff
@@ -221,20 +223,164 @@ var microprofileConfigCallBack = (function() {
         }
     };
 
-    var __saveServerEnvButton = function(event) {
+    var __refreshBrowserButton = function(event) {
+        if (event.type === "click" ||
+           (event.type === "keypress" && (event.which === 13 || event.which === 32))) {
+           // Click or 'Enter' or 'Space' key event...
+           contentManager.refreshBrowser(stepContent.getCurrentStepName());
+        }
+    };
+
+    var __saveButton = function(event) {
         if (event.type === "click" ||
            (event.type === "keypress" && (event.which === 13 || event.which === 32))) {
             // Click or 'Enter' or 'Space' key event...
             contentManager.saveEditor(stepContent.getCurrentStepName());
         }
     };
+   
+    //var injectConfigDefault = "download-url=ftp://music.com/us/download";
+
+    var __getInjectionConfigContent = function(content) {
+        var editorContents = {};
+        try {
+            // match
+            // private Config config;
+            //   <space or newline here>
+            // @Inject @ConfigProperty(name=\"download-url\", defaultValue=\"ftp://music.com/us/download\")
+            // private String downloadUrl;
+            var contentToMatch = "[\\s\\S]*private Config config;\\s*@Inject\\s*@ConfigProperty\\s*\(([\\s\\S]*)\)\\s*private String downloadUrl;";
+            var regExpToMatch = new RegExp(contentToMatch, "g");
+            var groups = regExpToMatch.exec(content);
+            
+            var params = groups[1];
+            params = params.replace('\n','');
+            params = params.replace(/\s/g, ''); // Remove whitespace
+            if (params.trim() !== "") {
+                params = params.split(',');
+            } else {
+                params = [];
+            }
+            editorContents.annotationParams = params;           
+        }
+        catch (e) {
+
+        }
+        return editorContents;
+    };
 
-    var __serverEnvRefreshBrowserButton = function(event) {
+    /*
+      Match the parameters. Returns
+        0 for no match
+        1 for exact match
+        2 for extra parameters
+    */
+    var __isParamInAnnotation = function(annotationParams, paramsToCheck) {
```

Comment:
I know you're still working on the validation part but you could make logic more simple as the content is not like circuitBreaker where annotation parameter was added one at a time for each step.